mkt/lookup/views.py

```diff
@@ -260,7 +260,10 @@ def transaction_refund(request, tx_uuid):
 
 @permission_required([('AppLookup', 'View')])
 def app_summary(request, addon_id):
-    app = get_object_or_404(Webapp.with_deleted, pk=addon_id)
+    query = {'app_slug': addon_id}
+    if unicode(addon_id).isdigit():
+        query = {'pk': addon_id}
```

Comment:
Personally I'd rather see this as an `if/else`. I was already confused if it could be both but it's really one or the other, which an `if/else` will help clear up.